examples/CMSapp/CMSapp.cpp

```diff
@@ -0,0 +1,266 @@
+             && "Volumes names mismatch");
+      visitAndSetMCindex(g4pvol->GetLogicalVolume(), pvol->GetLogicalVolume());
+    }
+  };
+
+  visitAndSetMCindex(g4world->GetLogicalVolume(), world->GetLogicalVolume());
+  std::cout << "Visited " << nphysical << " matching physical volumes\n";
+  return mcc_index;
+}
+
+void InitBVH()
+{
+  vecgeom::cxx::BVHManager::Init();
+  vecgeom::cxx::BVHManager::DeviceInit();
+}
+
+void PrintScoringPerVolume(const vecgeom::VPlacedVolume *placed, const ScoringPerVolume *scoring, int level = 0)
+{
+  for (auto *daughter : placed->GetDaughters()) {
+    std::cout << std::setw(level * 2) << "";
+    auto id = daughter->id();
+    std::cout << " ID " << id << " Charged-TrakL " << scoring->chargedTrackLength[id] / copcore::units::mm
+              << " mm; Energy-Dep " << scoring->energyDeposit[id] / copcore::units::MeV << " MeV" << std::endl;
+    PrintScoringPerVolume(daughter, scoring, level + 1);
+  }
+}
+
+int main(int argc, char* argv[])
+{
+  // Only outputs are the data file(s)
+  // Separate for now, but will want to unify
+  OPTION_STRING(gdml_file, "cms2018.gdml");
+  OPTION_INT(cache_depth, 0); // 0 = full depth
+  OPTION_INT(particles, 100);
+  OPTION_DOUBLE(energy, 10); // entered in GeV
+  energy *= copcore::units::GeV;
+  OPTION_INT(batch, -1);
+
+  vecgeom::Stopwatch timer;
+  timer.Start();
+  
+  // Initialize Geant4
+  auto g4world = InitGeant4(gdml_file);
+  if (!g4world) return 3;
+
+  // Initialize VecGeom
+  std::cout << "reading " << gdml_file << " transiently on CPU for VecGeom ...\n"; 
+  auto world = InitVecGeom(gdml_file, cache_depth);
+  if (!world) return 3;
+  
+  // Construct and initialize the G4HepEmState data/tables
+  std::cout << "initializing G4HepEm state ...\n";
+  G4HepEmState hepEmState;
+  InitG4HepEmState(&hepEmState);
+
+  // Initialize G4HepEm material-cut couple array indexed by VecGeom volume id.
+  // (In future we should connect the index directly to the VecGeom logical volume)
+  std::cout << "initializing material-cut couple indices ...\n";
+  int *MCCindex = CreateMCCindex(g4world, world, hepEmState);
+
+  // Load and synchronize the geometry on the GPU
+  std::cout << "synchronizing VecGeom geometry to GPU ...\n";
+  auto &cudaManager = vecgeom::cxx::CudaManager::Instance();
+  cudaManager.LoadGeometry(world);
+  cudaManager.Synchronize();
+  InitBVH();
+
+  auto time_cpu = timer.Stop();
+  std::cout << "Initialization took: " << time_cpu << " sec\n";
+
+  int NumVolumes = vecgeom::GeoManager::Instance().GetRegisteredVolumesCount();
+  int NumPlaced  = vecgeom::GeoManager::Instance().GetPlacedVolumesCount();
+
+  // Scoring is done per placed volume (for now...)
+  double *chargedTrackLength = new double[NumPlaced];
+  double *energyDeposit      = new double[NumPlaced];
+  ScoringPerVolume scoringPerVolume;
+  scoringPerVolume.chargedTrackLength = chargedTrackLength;
+  scoringPerVolume.energyDeposit      = energyDeposit;
+  GlobalScoring globalScoring;
```

Comment:
@jonapost and me had thought about scoring per volume. That would drastically reduce the number of numbers to output and compare and gives a reasonable abstraction that can be checked against Geant4.